batch2/batch/driver/scheduler.py

```diff
@@ -39,57 +81,75 @@ def __init__(self, app):
                 changed.clear()
 
     async def cancel_1(self):
-        records = self.db.execute_and_fetchall(
-            '''
-SELECT jobs.job_id, jobs.batch_id, cores_mcpu, instance_name
-FROM jobs
-INNER JOIN batches ON batches.id = jobs.batch_id
-INNER JOIN attempts ON jobs.batch_id = attempts.batch_id AND jobs.job_id = attempts.job_id AND jobs.attempt_id = attempts.attempt_id
-WHERE jobs.state = 'Running' AND (NOT jobs.always_run) AND batches.closed AND batches.cancelled
-LIMIT 50;
-''')
+        users = await self.get_user_resources()
```

Comment:
You just want users with running jobs here.  As you can tell from my style, I'd just put the query inline here.